torch/onnx/_internal/fx/symbolic_exporter.py

```diff
@@ -266,12 +195,16 @@ def export_without_parameters_and_buffers(
     # and we don't want
     #  ModeoProto.graph.input =
     #   [input_0, weight_0, input_1, weight_1, ..., input_n, weight_0, weight_1, ..., weight_m]
-    _move_placeholder_to_front(graph_module)
+    graph_module = passes.MovePlaceholderToFront(graph_module).run()
     # To save memory, move get_attr to input so that the generated model doesn't
-    # have weigh tensors. "replaced_attrs" are the list of replaced weight tensors.
-    replaced_attrs = _replace_get_attr_with_placeholder(graph_module)
+    # have weigh tensors. "replaced_attrs" are a tuple of replaced weight tensors.
+    replace_get_attr_with_placeholder_pass = passes.ReplaceGetAttrWithPlaceholder(
+        graph_module
+    )
+    graph_module = replace_get_attr_with_placeholder_pass.run()
+    replaced_attrs = replace_get_attr_with_placeholder_pass.replaced_attrs
     # Move all newly created placeholder nodes to the front of the graph.
-    _move_placeholder_to_front(graph_module)
+    graph_module = passes.MovePlaceholderToFront(graph_module).run()
```

Comment:
What happens if run was called twice?

Reply:
No thing happens in 2nd run. It will be good to add a test.

Reply:
Can't guarantee for all passes though. I kinda want to make it functional, but it seems there are cases where the state is useful. For now I don't worry too much about this, but let me know if you have particular concern.

Reply:
And at least diagnostics catches it running twice, and the callstack that leads to that.

Reply:
Could add a guard in the class against multi run too if we really hate it.

Reply:
I am not really against multiple runs. For iterative passes we might actually want it? Just trying to clarify and understand the behavior